timeseries/src/autogluon/timeseries/utils/features.py

```diff
@@ -284,3 +300,53 @@ def _check_required_columns_are_present(
             raise ValueError(
                 f"{len(missing_columns)} columns are missing from {data_frame_name}: {reprlib.repr(missing_columns.to_list())}"
             )
+
+
+class AbstractFeatureImportanceTransform:
+    """Abstract class for transforms that replace a given feature with dummy or shuffled values,
+    for use in feature importance operations.
+    """
+
+    def __init__(
+        self,
+        covariate_metadata: CovariateMetadata,
+        **kwargs,
+    ):
+        self.covariate_metadata: CovariateMetadata = covariate_metadata
+
+    def _apply_transform(self, data: TimeSeriesDataFrame, feature_name: str) -> TimeSeriesDataFrame:
+        raise NotImplementedError
+
+    def transform(self, data: TimeSeriesDataFrame, feature_name: str, **kwargs) -> TimeSeriesDataFrame:
+        if feature_name not in self.covariate_metadata.all_features:
+            raise ValueError(f"Target feature {feature_name} not found in covariate metadata")
+
+        # feature transform works on a shallow copy of the main time series data frame
+        # but a deep copy of the static features.
+        data = data.copy(deep=False)
+
+        return self._apply_transform(data, feature_name, **kwargs)
```

Comment:
This is similar to my initial implementation of feature importance in Tabular, but it turns out to be compute inefficient as it does excessive data mutations (I think). If you run into performance issues, consider checking the Tabular code implementation I linked elsewhere in my review.

Reply:
Thanks! I worked on this a little and so far it looks good because we take care of shallow copies in the custom `TimeSeriesDataFrame`. I'll keep this in mind.